lib/scholar/metrics.ex

```diff
@@ -332,28 +351,58 @@ defmodule Scholar.Metrics do
       iex> y_pred = Nx.tensor([0, 1, 0, 2, 2, 2], type: {:u, 32})
       iex> Scholar.Metrics.confusion_matrix(y_true, y_pred, num_classes: 3)
       #Nx.Tensor<
-        u64[3][3]
+        f32[3][3]
+        [
+          [1.0, 1.0, 0.0],
+          [1.0, 0.0, 1.0],
+          [0.0, 0.0, 2.0]
+        ]
+      >
+
+      iex> y_true = Nx.tensor([0, 0, 1, 1, 2, 2], type: {:u, 32})
+      iex> y_pred = Nx.tensor([0, 1, 0, 2, 2, 2], type: {:u, 32})
+      iex> sample_weights = [2, 5, 1, 1.5, 2, 8]
+      iex> Scholar.Metrics.confusion_matrix(y_true, y_pred, num_classes: 3, sample_weights: sample_weights, normalize: :predicted)
+      #Nx.Tensor<
+        f32[3][3]
         [
-          [1, 1, 0],
-          [1, 0, 1],
-          [0, 0, 2]
+          [0.6666666865348816, 1.0, 0.0],
+          [0.3333333432674408, 0.0, 0.1304347813129425],
+          [0.0, 0.0, 0.8695651888847351]
         ]
       >
   """
   deftransform confusion_matrix(y_true, y_pred, opts \\ []) do
-    confusion_matrix_n(y_true, y_pred, NimbleOptions.validate!(opts, @general_schema))
+    confusion_matrix_n(y_true, y_pred, NimbleOptions.validate!(opts, @confusion_matrix_schema))
```

Comment:
If you want, you can make weights default to an integer if both inputs are integers, is that what you want?

Reply:
Ideally, I would make floats always if weights are passed (but this is impossible). So yes, if both data and weights are integers, then I want them to preserve the type.

Reply:
Oh, but actually we can do this in deftransform